### PR TITLE
Some tasks are now only using module manifest version number

### LINKED ISSUE #160: Only tasks that need to initiate (set) should evaluate module version
Currently all tasks are evaluating the module version using either existing ModuleVersion, GitVersion or the module manifest. In certain scenarios this gets messed up.

For example if the a version is built, e.g. 0.105.1 and then opening another session and running the task `test` it is possible that the version evaluates differently.

```
ModuleVersion = 0.105.2-preview0001
CodeCoverage  = C:\source\Sampler\output\Sampler\0.105.1\DSCResources\DSC_Folder\DSC_Folder.psm1
```

In this case the task `test` (task `Invoke_pester_tests`) evaluated ModuleVersion using gitversion (thinks module version is 0.105.2) because gitversion saw another commit that would raise the patch version again. But the `CodeCoverage` property is using the imported module in the output folder to evaluate the path (using the `ModuleBase` property) which have the version as was set in the module manifest (0.105.1). 

I suggest we change the tasks that does not need to initiate (set) the version in the module manifest to not evaluate but instead always use the module version from the module manifest in the output folder.

### PR DESCRIPTION
**Pull request by johlju**
# Pull Request

- Fixes #160

### Changed
- Tasks that are not setting the version in the module manifest or otherwise
  need to evaluate the module version are now using the the module version
  from the nodule manifest that the build task evaluated and set (issue #160).

### Fixed

- Fixed so module version can be detected using GitVersion on macOS and
  Linux.

## Task list

- [x] The PR represents a single logical change. i.e. Cosmetic updates should go in different PRs.
- [x] Added an entry under the Unreleased section of in the CHANGELOG.md as per [format](https://keepachangelog.com/en/1.0.0/).
- [x] Local clean build passes without issue or fail tests (`build.ps1 -ResolveDependency`).
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [ ] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [ ] Unit tests added/updated. See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] Integration tests added/updated (where possible). See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [x] New/changed code adheres to [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md).

---
This change is [Reviewable](https://reviewable.io/reviews/gaelcolas/sampler/162)

### CODE REVIEW

**Comment by johlju** [comment]
Thinking more about about the change to Get-ModuleVersion I think we should split it up into two functions; `Get-BuildVersion` and `Get-ModuleVersion`.

- `Get-BuildVersion`: Evaluates the build version from `property` (Parameter, parent scope, environment varaible), GitVersion or source module manifest.
- `Get-ModuleVersion`: Gets the module version from the built module's module manifest.

**Comment by johlju** [comment]
Ready for review.

**Comment by johlju** [comment]
@ykuijs you think this change with the issue you are seeing in SharePointDsc?

**Comment by ykuijs** [comment]
I think this makes perfect sense!

File: .build/tasks/Common.Functions.psm1 [hunk 1]
```diff
@@ -130,6 +117,45 @@ function Get-ModuleVersion
     return $moduleVersionParts.ModuleVersion
 }
 
+function Get-ModuleVersion
```
Comment (gaelcolas) [inline .build/tasks/Common.Functions.psm1:120]: non-blocking:
Get-BuiltModuleVersion?
I'm just afraid Get-ModuleVersion might be so generic that it conflicts with another command one day...

Comment (johlju) [reply .build/tasks/Common.Functions.psm1:120]: Yes, you are right. Fixing.

**Comment by johlju** [comment]
@ykuijs I will release this as soon as the pipeline finishes. Let me know if this does not resolve the issue we were seeing with the builds and I help debug further.

I will also fix DscResource.DocGenerator as soon as Sampler is released.

**Pull request merged**